### ISSUE TITLE
Prevent all subtypes

### ISSUE THREAD

**Issue by damien-git** [blueriver/MuraCMS#2757]
I would like to prevent users from inserting pages under links (I wonder why this is allowed by default).

If I create a Link/Default class extension, set "Allow users to add only specific subtypes?" to Yes, and avoid selecting any subtype, my choice is not saved at all. Also, if I select something, save, then unselect it and save again, the second save is not taken into account.

I have encountered this issue before, but I am not seeing any report of it, so I guess I did not report it. Is this a bug ?

**Comment by mattlevine** [comment]
Perhaps you could use onBeforeContentSave event

```
function onBeforeContentSave(m){
    var content=m.content();
    if(content.hasParent() && content.getParent().getType() == 'Link'){
    content.getErrors().parentIsLink='You can not add a content node under a link";
   }
}
```

Having a more elegant approach to this is in our future goals

**Comment by damien-git** [comment]
Thanks. I tried that, but content.hasParent() is false when onBeforeContentSave() is called. Another issue with this workaround is that onBeforeContentSave() is only called when someone saves the page, so typically after entering content.

**Comment by mattlevine** [comment]
Yes, that's why I mention that one of our future goals is to have a more elegant solution.  Keep in mind that they user will be presented with an error method with the content preserved.  They could then got to the "publishing" tab and select a new content parent.
